news/8042.feature

```diff
@@ -0,0 +1,15 @@
+Add CLI options for enhancing requests with HTTP headers
+
+-H, --header <key:val>      HTTP header to include in all requests. This option
+                            can be used multiple times. Conflicts with
+                            --extra-index-url.
+
+Example:
+
+```
+pip install \
+  --index-url http://pypi.index/simple/ \
+  --trusted-host pypi.index \
+  -H 'X-Spam: ~*~ SPAM ~*~' \
+  requests
+```
```

Comment:
The changelog is probably not a good place to put such detailed description to a feature addition. I would add something to the documentation, and just say something like *see documentation for more details* here.